OOInteraction/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -0,0 +1,155 @@
+
+	if (textSoFar.trimmed().startsWith("add ", Qt::CaseInsensitive))
+	{
+		auto nodeName = textSoFar.trimmed().mid(4);
+		auto names = findNames(nodeName.split("_"),
+						Model::AllTreeManagers::instance().loadedManagers().first()->root());
+		//Shorter names usually have less parts to the fully qualified name -> suggest them first
+		std::sort(names.begin(), names.end(), [](QString first, QString second)
+											{ return first.length() < second.length(); });
+		//Limit the number of suggestions
+		names = names.mid(0, 10);
+		for (auto name : names)
+			suggestions.append(new Interaction::CommandSuggestion("add " + name,
+									"Add node " + name + " to the view"));
+	}
+	else if (QString("add ").startsWith(textSoFar.trimmed(), Qt::CaseInsensitive) )
+			suggestions.append(new Interaction::CommandSuggestion("add ", "Add nodes to the current view"));
+
+	return suggestions;
+}
+
+QStringList CAddNodeToViewByName::findNames(QStringList nameParts, Model::Node* root)
+{
+	auto declaration = DCast<OOModel::Declaration>(root);
+	QStringList result;
+	//If it isn't a declaration, just pass it on
+	if (!declaration)
+		for (auto child : root->children())
+			result.append(findNames(nameParts, child));
+	//If it is a declaration, and the name matches, take the part of name it matches and search for the rest
+	else if (nameParts.size() > 0 && declaration->name().startsWith(nameParts[0]))
+	{
+		for (auto child : root->children())
+			for (auto name : findNames(nameParts.mid(1), child))
+				result.append(declaration->name() + "_" + name);
+		if (nameParts.size() == 1)
+			result.append(declaration->name());
+	}
+	//If we don't have any name, accept any declarations for suggestions
+	else if (nameParts.size() == 0 && isSuggestable(declaration))
+	{
+		for (auto child : root->children())
+			for (auto name : findNames(nameParts, child))
+				result.append(declaration->name() + "_" + name);
+		result.append(declaration->name());
+	}
+	return result;
+}
+
+OOModel::Declaration* CAddNodeToViewByName::findNode(QString fullyQualifiedName, Model::Node* root)
+{
+	//TODO can't use "." for some reason (command unknown..)
+	auto nameParts = fullyQualifiedName.split("_");
+	QList<Model::Node*> toSearch{root};
+	int currentPart = 0;
+	while (!toSearch.isEmpty() && currentPart < nameParts.size())
+	{
+		auto current = toSearch.takeLast();
+		auto declaration = DCast<OOModel::Declaration>(current);
+		if (declaration && declaration->name() == nameParts[currentPart]
+				&& currentPart == nameParts.size() - 1)
+			return declaration;
+		//If it is not a declaration -> search in children.
+		//Else only continue the path if the name fits
+		if (!declaration)
+			toSearch.append(current->children());
+		else if (declaration->name() == nameParts[currentPart])
+		{
+			toSearch.clear();
+			toSearch.append(current->children());
+			currentPart++;
+		}
+	}
+	return nullptr;
+}
+
+bool CAddNodeToViewByName::isSuggestable(OOModel::Declaration *declaration)
```

Comment:
To achieve something similar, using just generic concepts (without relying on OO specific) use the `node->symbolType` and check for method and container.